### PR TITLE
Use fs.rm() instead of rimraf

### PR DESCRIPTION
**Pull request by kevinoid**
A regression occurred in 0.2.2 which can be reproduced using:

```js
const assert = require('assert');
const tmp = require('tmp');
tmp.dir({ unsafeCleanup: true }, (err, path, cleanup) => {
  assert.ifError(err);
  cleanup(assert.ifError);
});
```
This works with 0.2.1.  With 0.2.2 it fails with:

    /path/to/tmp/node-tmp/lib/tmp.js:358
            return removeFunction(fileOrDirName, next || function() {});
                   ^

    TypeError: removeFunction is not a function
        at _cleanupCallback (/path/to/tmp/node-tmp/lib/tmp.js:358:16)
        at /path/to/tmp/node-tmp/repro.js:5:3
        at _dirCreated (/path/to/tmp/node-tmp/lib/tmp.js:207:7)
        at FSReqCallback.oncomplete (node:fs:192:23)

This occurs because 00bb5b282faadfd45a4e76d3e05591ab42c71027 upgraded the rimraf dependency from ^3.0.0 to ^5.0.5 without handling the change to a `Promise`-based API in 4.0.0 (https://github.com/isaacs/rimraf/commit/a71e7f9afb2d508496d32186daa3126bff9b5ef6) or the removal of the default export in 5.0.0 (https://github.com/isaacs/rimraf/commit/c7a3fd4e7e299eb5ba27d913e1d4d04951eb661d).

This commit fixes the issue by dropping the `rimraf` dependency in favor of `fs.rm({recursive: true})`.

Thanks for considering,
Kevin

Fixes: #295

### CODE REVIEW

**Review by raszi — APPROVED** [review]
Looks good! Thank you very much!

**Comment by raszi** [comment]
I believe the tests need to be updated as well.

**Comment by kevinoid** [comment]
My pleasure.  Thanks for reviewing it @raszi.

> I believe the tests need to be updated as well.

Whoops!  Right you are.  Sorry I missed that.

I've pushed a new version which updates the tests, `package-lock.json`, and the docs.  Let me know if I've missed anything else.

P.S. How do you feel about support for Node.js 14.0 to 14.14 (when `fs.rm()` was added)?  I have a fallback from `fs.rm()` to `fs.rmdir()` in `lib/tmp.js` to support <14.14, but didn't bother in the tests, since CI doesn't run it.  Would it make sense to drop support for Node.js < 14.14, or to test it and add a CI job?

**Comment by raszi** [comment]
I don't mind dropping that it isn't supported anymore:

https://nodejs.org/en/about/previous-releases

image

**Comment by kevinoid** [comment]
Sounds good to me.  I've dropped the `fs.rmdir` fallback to simplify the code a bit.  Let me know if there are any other changes you'd like.

**Comment by kevinoid** [comment]
Note: I just pushed an update to generate `package-lock.json` with npm from Node 14 to fix the CI error.

File: package.json [hunk 1]
```diff
@@ -25,7 +25,6 @@
     "node": ">=14"
```
Comment (raszi) [inline package.json:25]: I believe this should be also updated:

https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines

Comment (kevinoid) [reply package.json]: Good idea!  Done.

**Review by raszi — APPROVED** [review]
Nice work! Thank you very much!

**Pull request merged**